### PR TITLE
Add support for syscall waitid

### PR DESCRIPTION
**Pull request by Sahakait**

### CODE REVIEW

File: kernel/syscall.c [hunk 1]
```diff
@@ -160,6 +160,7 @@ static pair_t _pairs[] = {
     {SYS_execve, "SYS_execve"},
     {SYS_exit, "SYS_exit"},
     {SYS_wait4, "SYS_wait4"},
+    {SYS_waitid, "SYS_waitid"},
```
Comment (vtikoo) [inline kernel/syscall.c:163]: Theres already an entry for `SYS_waitid` in `_pairs`.

File: include/myst/thread.h [hunk 2]
```diff
@@ -397,6 +397,13 @@ pid_t myst_generate_tid(void);
 
 pid_t myst_gettid(void);
 
+long myst_wait(
+    pid_t pid,
+    int* wstatus,
+    siginfo_t* infop,
+    int options,
+    struct rusage* rusage);
```
Comment (vtikoo) [inline include/myst/thread.h:405]: Import is missing for `rusage`'s definition.

File: kernel/thread.c [hunk 3]
```diff
@@ -377,10 +377,72 @@ long myst_syscall_wait4(
...
+
+long myst_syscall_waitid(
+    idtype_t idtype,
+    id_t id,
+    siginfo_t* infop,
+    int options)
+{
+    long ret = 0;
+
+    if (options & ~(WEXITED | WSTOPPED | WCONTINUED | WNOHANG))
```
Comment (vtikoo) [inline kernel/thread.c:394]: WNOWAIT is missing here. And also some Linux specific additional flags - __WNOTHREAD, __WALL and __WCLONE.

Comment (Sahakait) [reply kernel/thread.c:394]: I need to confirm with Mike again, but we will probably end up only supporting WEXITED and WNOHANG.

**Review by vtikoo — APPROVED** [review]
Looks good wrt dotnet's usage of `waitid`.

**Pull request merged**